app/lib/frontend/backend.dart

```diff
@@ -239,47 +238,31 @@ class Backend {
     final inviteId = models.PackageInvite.createId(type, recipientEmail);
     final pkgKey = db.emptyKey.append(models.Package, id: packageName);
     final inviteKey = pkgKey.append(models.PackageInvite, id: inviteId);
-    return await db.withTransaction((tx) async {
-      final list = await tx.lookup<models.PackageInvite>([inviteKey]);
-      final invite = list.single;
+    final invite = (await db.lookup<models.PackageInvite>([inviteKey])).single;
 
-      // Invite entry does not exists.
-      if (invite == null) {
-        await tx.rollback();
-        return null;
-      }
+    if (invite == null) {
+      return null;
+    }
 
-      // Consistency check
-      if (invite.recipientEmail != recipientEmail) {
-        await tx.rollback();
-        return null;
-      }
+    if (invite.isValid(recipientEmail: recipientEmail, urlNonce: urlNonce)) {
+      return invite;
+    }
+    return null;
+  }
 
-      // Invite entry has expired.
-      if (invite.isExpired()) {
-        tx.queueMutations(deletes: [inviteKey]);
+  /// Updates the confirmed timestamp in the [invite].
+  Future confirmPackageInvite(models.PackageInvite invite) async {
+    await db.withTransaction((tx) async {
+      final pi = (await tx.lookup<models.PackageInvite>([invite.key])).single;
+      if (pi != null && pi.confirmed == null) {
+        pi.confirmed = DateTime.now().toUtc();
+        tx.queueMutations(inserts: [pi]);
         await tx.commit();
```

Comment:
I might be missing something, but don't you also need to update the `Package` entity to add the `userId` to `uploaders`?

Reply:
Good catch, updated the PR, PTAL.